### PR TITLE
feat: check for blank/missing agency_id

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR provides changes to verify:
- agency_id is provided in all cases (WARNING if `agency.txt` contains an unique row)
- agency_id is not blank when provided (WARNING)

 **New notices**
 - AgencyIdBlankNotice
 - AgencyIdMissingNotice
 
**Expected behavior:** 

A notice should be issued if:
- agency_id is blank
- agency_id is not provided when `agency.txt`  only contains 1 row

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java [hunk 1]
```diff
@@ -48,19 +52,31 @@
...
+          noticeContainer.addValidationNotice(new AgencyIdBlankNotice(agency.csvRowNumber()));
+        }
+      }
       return;
     }
 
     for (GtfsAgency agency : agencyTable.getEntities()) {
-      // agency_id is required when there are 2 or more agencies.
       if (!agency.hasAgencyId()) {
+        // agency_id is required when there are 2 or more agencies.
```
[the remark this answers is not in the document]
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java]: > Actually - what is the current behavior when parsing whitespace? Is this detected when parsing the field? I recall a check for newlines in fields as an error but don't recall if whitespace is already checked when parsing.

There is logic to nullify empty strings (i.e `""`): https://github.com/MobilityData/gtfs-validator/blob/bc8c8e35d526c1dd896bf9abed365355fd3a0434/core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/CsvRow.java#L81

But the case where a field is made of whitespaces is not currently handled if I'm correct.

**Review by maximearmstrong — COMMENTED** [review]
LGTM

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java [hunk 2]
```diff
@@ -94,4 +110,16 @@ public void validate(NoticeContainer noticeContainer) {
...
+  /**
+   * Checks is an `agency.agency_id` is missing or blank i.e. `agency_id` is a sequence of
+   * whitespaces
+   *
+   * @param agency the {@code GtfsAgency} to check the id
+   * @return true if `agency.agency_id` is missing or blank (i.e. `agency_id` is a sequence of
+   *     whitespaces), false otherwise
+   */
+  private boolean isAgencyIdMissingOrBlank(GtfsAgency agency) {
+    return !agency.hasAgencyId() || StringUtils.isBlank(agency.agencyId());
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java:123]: `StringUtils.isBlank(agency.agencyId())` will be always false. I am leaving to you as an exercise to figure out why this is so.

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidatorTest.java [hunk 3]
```diff
@@ -227,4 +228,84 @@ public void agenciesWithOmittedLanguageShouldNotGenerateNotice() {
...
+  public void multipleAgencyWithBlankIdShouldGenerateNotice() {
+    NoticeContainer noticeContainer = new NoticeContainer();
+    AgencyConsistencyValidator underTest = new AgencyConsistencyValidator();
+    underTest.agencyTable =
+        createAgencyTable(
+            noticeContainer,
+            ImmutableList.of(
+                createAgency(
+                    1,
+                    "    ",
```
Comment (aababilov) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidatorTest.java:284]: This is not realistic, it may never be " ".

**Review by aababilov — CHANGES_REQUESTED** [review]
Please could you provide a sample dataset where the existing code is giving incomplete or misleading information and where the suggested code is producing a significantly better validation report?

**Comment by lionel-nj** [comment]
Closing. We will not continue work on this PR since the parsing process makes this use case not relevant: blank and empty Strings are nullified so that a `GtfsAgency` will never be created with `agency.agency_id = "   "`

**Comment by lionel-nj** [comment]
Closing. We will not continue work on this PR since the parsing process makes this use case not relevant: blank and empty Strings are nullified so that a GtfsAgency will never be created with agency.agency_id = " "

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java (hunk 2 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/AgencyConsistencyValidator.java:123]: https://github.com/MobilityData/gtfs-validator/pull/799#issuecomment-799704893

**Pull request closed without merging**